pkg/skaffold/kubernetes/status/status_check.go

```diff
@@ -71,40 +75,65 @@ type Config interface {
 	GetNamespaces() []string
 	StatusCheckDeadlineSeconds() int
 	Muted() config.Muted
-}
-
-// Checker waits for the application to be totally deployed.
-type Checker interface {
-	Check(context.Context, io.Writer) error
+	StatusCheck() (*bool, error)
 }
 
 // statusChecker runs status checks for pods and deployments
-type statusChecker struct {
+type Checker struct {
 	cfg             Config
 	labeller        *label.DefaultLabeller
 	deadlineSeconds int
 	muteLogs        bool
+	seenResources   resource.Group
+	singleRun       singleflight.Group
```

Comment:
for my own learning, what's the difference between this and `sync.Once`?

Reply:
`singleflight.Group` runs a single instance for `n` _concurrent_ requests. But it'll rerun if there's another series of requests following the last concluded run. On the other hand, `sync.Once` will only have 1 run per instantiation.

The use of `singleflight.Group` here helps to only run one k8s checker per call to the `checker_mutex.Check` method which multiplexes calls from its constituent `Checker` instances.